src/optimize.rs

```diff
@@ -453,8 +453,15 @@ impl<I: Iterator<Item = Segment>> Iterator for Optimizer<I> {
 }
 
 /// Computes the total encoded length of all segments.
-pub fn total_encoded_len(segments: &[Segment], version: Version) -> usize {
-    segments.iter().map(|seg| seg.encoded_len(version)).sum()
+pub fn total_encoded_len<IntoIter, Item>(segments: IntoIter, version: Version) -> usize
+where
+    IntoIter: IntoIterator<Item = Item>,
+    Item: std::borrow::Borrow<Segment>,
```

Comment:
I did look at the comment thread you've pointed out.  Is there any action item for me here?

AFAIU the `Clone` aspect doesn't apply here - `total_encoded_len` doesn't need to clone the iterator nor the items.

Reply:
> I did look at the comment thread you've pointed out. Is there any action item for me here?

I was referring in particular to the possibility to remove the generic on Item because you can refer to it via `IntoIter::Item` in the where clause, such this (note only `I` as generic):

```
/// Computes the total encoded length of all segments.
pub fn total_encoded_len<I>(segments: I, version: Version) -> usize
where
    I: IntoIterator,
    I::Item: Borrow<Segment>,
{
    segments
        .into_iter()
        .map(|seg| seg.borrow().encoded_len(version))
        .sum()
}
```

Reply:
Oh, I see.  Thanks for explaining.  Done.